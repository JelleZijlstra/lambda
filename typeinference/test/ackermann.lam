let rec ackermann = \m. \n. if m = 0
	then n + 1
	else if n = 0
		then ackermann (m - 1) 1
		else ackermann (m - 1) (ackermann m (n - 1)) in

let x = print (ackermann 3 6)
in
x