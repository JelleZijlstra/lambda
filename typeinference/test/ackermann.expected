%%stdout%%
509
Result: module
	let __result__ = ()
end
%%stderr%%
